src/Behat/Context/CookieComplianceContext.php

```diff
@@ -111,47 +109,107 @@ class CookieComplianceContext extends RawMinkContext {
    *   List of third party domains that loads cookies to add.
    */
   public function __construct(
-    string $cookie_agree_selector,
-    string $cookie_banner_selector,
-    array $cookies,
+    string $cookie_manager_type = '',
+    string $cookie_agree_selector = '',
+    string $cookie_reject_selector = '',
+    string $cookie_banner_selector = '',
+    array $cookies = [],
     array $cookies_ignored = [],
     array $cookies_third_party_domains_ignored = [],
-    array $cookies_third_party_domains_included = []
+    array $cookies_third_party_domains_included = [],
   ) {
-    $this->cookieAgreeSelector = $cookie_agree_selector;
-    $this->cookieBannerSelector = $cookie_banner_selector;
+    switch ($cookie_manager_type) {
+      case 'onetrust':
+        $this->cookieManager = new OneTrustCookieManager($cookie_agree_selector, $cookie_reject_selector, $cookie_banner_selector);
+        break;
+
+      default:
+        $this->cookieManager = new DefaultCookieManager($cookie_agree_selector, $cookie_reject_selector, $cookie_banner_selector);
+    }
     $this->cookies = $cookies;
     $this->cookiesIgnored = $cookies_ignored;
     $this->cookiesThirdPartyDomainsIgnored = $cookies_third_party_domains_ignored;
     $this->cookiesThirdPartyDomainsIncluded = $cookies_third_party_domains_included;
   }
 
   /**
-   * Accept cookie compliance.
-   *
-   * It accepts it by accepting all the categories
-   * that are accepted by default.
+   * Accept cookie compliance by clicking the button.
    *
    * @Then I accept cookies
    */
   public function iAcceptCookies() {
-    $agree_button = $this->getSession()->getPage()->find('css', $this->cookieAgreeSelector);
-    if ($agree_button instanceof NodeElement) {
-      // Some cookie banners have animations that do not let click the agree
+    $this->handleCookieBanner(
+        $this->cookieManager->getAcceptButtonSelector(),
+        $this->cookieManager->getCookieBannerSelector(),
+        'accept'
+    );
+  }
+
+  /**
+   * Accept cookies automatically.
+   *
+   * @BeforeScenario @cookies-accepted
+   */
+  public function acceptCookiesBeforeScenario() {
+    $this->visitPath('/');
+    $this->cookieManager->acceptCookies($this->getSession());
+  }
+
+  /**
+   * Reject cookie compliance by clicking the button.
```

Comment:
Improved comment: "Reject cookies  by clicking the reject button int the cookie manager popup or banner.".